com.unity.multiplayer.mlapi/Runtime/Timing/NetworkTimeSystem.cs

```diff
@@ -0,0 +1,111 @@
+        /// </summary>
+        public double ServerBuffer { get; set; }
+
+        /// <summary>
+        /// Gets or sets a threshold in seconds used to force a hard catchup of network time.
+        /// </summary>
+        public double HardResetThreshold { get; set; }
+
+        /// <summary>
+        /// Gets or sets the ratio at which the NetworkTimeSystem speeds up or slows down time.
+        /// </summary>
+        public double AdjustmentRatio { get; set; }
+
+        public double LocalTime => m_Time + m_CurrentLocalTimeOffset;
+
+        public double ServerTime => m_Time + m_CurrentServerTimeOffset;
+
+        //TODO This is used as a workaround to pass this back into the sync function will be removed once we get correct value.
+        internal double TimeSystemInternalTime => m_Time;
+
+        public NetworkTimeSystem(double localBuffer, double serverBuffer, double hardResetThreshold, double adjustmentRatio = 0.01d)
+        {
+            LocalBuffer = localBuffer;
+            ServerBuffer = serverBuffer;
+            HardResetThreshold = hardResetThreshold;
+            AdjustmentRatio = adjustmentRatio;
+        }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="NetworkTimeSystem"/> class for a server instance.
+        /// The server will not apply any buffer values which ensures that local time equals server time.
+        /// </summary>
+        /// <returns>The instance.</returns>
+        public static NetworkTimeSystem ServerTimeSystem()
+        {
+            return new NetworkTimeSystem(0, 0, double.MaxValue);
+        }
+
+        /// <summary>
+        /// Advances the time system by a certain amount of time. Should be called once per frame with Time.deltaTime or similar.
+        /// </summary>
+        /// <param name="deltaTime">The amount of time to advance. The delta time which passed since Advance was last called.</param>
+        /// <returns></returns>
+        public bool Advance(double deltaTime)
+        {
+            m_Time += deltaTime;
+
+            if (Math.Abs(m_DesiredLocalTimeOffset - m_CurrentLocalTimeOffset) > HardResetThreshold || Math.Abs(m_DesiredServerTimeOffset - m_CurrentServerTimeOffset) > HardResetThreshold)
+            {
+                m_Time += m_DesiredServerTimeOffset;
+
+                m_DesiredLocalTimeOffset -= m_DesiredServerTimeOffset;
+                m_CurrentLocalTimeOffset = m_DesiredLocalTimeOffset;
+
+                m_DesiredServerTimeOffset = 0;
+                m_CurrentServerTimeOffset = 0;
+
+                return true;
+            }
+
+            m_CurrentLocalTimeOffset += deltaTime * (m_DesiredLocalTimeOffset > m_CurrentLocalTimeOffset ? AdjustmentRatio : -AdjustmentRatio);
+            m_CurrentServerTimeOffset += deltaTime * (m_DesiredServerTimeOffset > m_CurrentServerTimeOffset ? AdjustmentRatio : -AdjustmentRatio);
+
+            return false;
+        }
+
+        public void Initialize(double serverTime, double rtt)
```

Comment:
nit: I'd say this is a reset function not an initialization function. Constructor does some basic initialization but this function resets things.
```suggestion
        public void Reset(double serverTime, double rtt)
```

Reply:
Yes you are right, I' m renaming this to `Reset`